lightningd/peer_control.c

```diff
@@ -1754,6 +1755,59 @@ static const struct json_command listpeers_command = {
 /* Comment added to satisfice AUTODATA */
 AUTODATA(json_command, &listpeers_command);
 
+static void json_add_scb(struct lightningd *ld,
+			  struct json_stream *response,
+			  struct channel *c)
+{
+	u8 *msg = towire_static_chan_backup(tmpctx, 
+										c->scb->id, 
+										&c->scb->cid, 
+										&c->scb->node_id, 
+										&c->scb->addr, 
+										&c->scb->funding);
+
+	json_add_hex_talarr(response, NULL,
+			msg);
+}
+
+/*This will return a SCB for all the channels currently loaded 
+	in the in-memory channel*/
+static struct command_result *json_hex_scb(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *obj UNNEEDED,
+					     const jsmntok_t *params)
+{
+	struct json_stream *response;
+
+	if (!param(cmd, buffer, params, NULL))
+        return command_param_failed();
+	response = json_stream_success(cmd);
+
+	json_array_start(response, "scb");
+
+	/* FIXME: Filter out some channels based on their state? */
+
+	struct peer *peer;
+	struct channel *channel;
+	list_for_each(&cmd->ld->peers, peer, list)
+		list_for_each(&peer->channels, channel, list)
+			json_add_scb(cmd->ld, response, channel);
+
+	json_array_end(response);
+
+	return command_success(cmd, response);
+}
+
+static const struct json_command hex_scb_command = {
+	"hex-scb",
```

Comment:
Please don't use `-` in command names, they translate very poorly into other languages, requiring workarounds and renaming. We usually just go with something like `hexscb`. Also it'd be nicer to have `staticbackup` as a name instead, `hex` just being the format, allowing us to iterate in the future, without requiring a rename.

Reply:
Maybe `static_backup` is better? it makes the docs more accessible for screen readers too.

But usually, the `_` will translate directly into other languages with the camel case, so it is more difficult for single word command name.

But it is minor if it requires a lot of work!